### PR TITLE
Bugs - Retest button broken link

### PR DESCRIPTION
**Pull request by jicata**

### CODE REVIEW

File: Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml [hunk 1]
```diff
@@ -83,4 +85,15 @@
...
     });
+    function createRetestForm(id) {
+        var token = $("[name='__RequestVerificationToken']").val();
+        var formOpeningTagHtml = "<form action=\"/Administration/Submissions/Retest\" class=\"inline-form\" method=\"post\" role=\"form\">";
+        var antiforgeryTokenHtml = "<input name=\"__RequestVerificationToken\" type=hidden value=\"" + token + "\">";
+        var inputHtml = "<input data-val=\"true\" data-val-number=\"The field Id must be a number.\" data-val-required=\"The Id field is required.\" id=\"Id\" name=\"Id\" type=\"hidden\" value=\"" + id + "\" />";
+        var submitHtml = "<input type=\"submit\" class=\"k-button k-button-icontext\" value=Retest />";
+        var formHtmlClosingTag = "</form>";
+        var formHtml = formOpeningTagHtml + inputHtml + antiforgeryTokenHtml + submitHtml + formHtmlClosingTag;
+        console.log(formHtml);
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml]: Do you need this? I think you can safely remove it.

File: Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml [hunk 2]
```diff
@@ -8,23 +10,23 @@
...
+    columns.ForeignKey(model => model.Status, (IEnumerable)ViewBag.SubmissionStatusData, "Id", "Name");
+    columns.Bound(model => model.Points);
+    columns.Bound(model => model.CreatedOn).Filterable(x => x.UI(GridFilterUIRole.DateTimePicker)).Hidden();
+    columns.Bound(model => model.ModifiedOn).Filterable(x => x.UI(GridFilterUIRole.DateTimePicker)).Hidden();
+    columns.Template(@<text></text>).ClientTemplate("<a href='" + Url.Action("Update", ControllerName) + "/#= Id #' class='k-button k-button-icontext'>" + GeneralResource.Change + "</a>").Title(GeneralResource.Change);
+    columns.Template(@<text></text>).ClientTemplate("<a href='" + Url.Action("Delete", ControllerName) + "/#= Id #' class='k-button k-button-icontext'>" + GeneralResource.Delete + "</a>").Title(GeneralResource.Delete);
+    columns.Template(@<text></text>).ClientTemplate("<a href='" + Url.Action("View", "Submissions", new { Area = "Contests" }) + "/#= Id #' class='k-button k-button-icontext' target='_blank'>" + Resource.Details + "</a>").Title(Resource.Details);  
+        columns.Template(@<text></text>).ClientTemplate("#=createRetestForm(Id)#").Title(Resource.Retest);
```
Comment (Innos) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml]: Move 1 Tabulation to the left.

File: Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml [hunk 3]
```diff
@@ -8,23 +10,23 @@
...
-        columns.Bound(model => model.CreatedOn).Filterable(x => x.UI(GridFilterUIRole.DateTimePicker)).Hidden();
-        columns.Bound(model => model.ModifiedOn).Filterable(x => x.UI(GridFilterUIRole.DateTimePicker)).Hidden();
-        columns.Template(@<text></text>).ClientTemplate("<a href='" + Url.Action("Update", ControllerName) + "/#= Id #' class='k-button k-button-icontext'>" + GeneralResource.Change + "</a>").Title(GeneralResource.Change);
-        columns.Template(@<text></text>).ClientTemplate("<a href='" + Url.Action("Delete", ControllerName) + "/#= Id #' class='k-button k-button-icontext'>" + GeneralResource.Delete + "</a>").Title(GeneralResource.Delete);
-        columns.Template(@<text></text>).ClientTemplate("<a href='" + Url.Action("View", "Submissions", new { Area = "Contests" }) + "/#= Id #' class='k-button k-button-icontext' target='_blank'>" + Resource.Details + "</a>").Title(Resource.Details);
-        columns.Template(@<text></text>).ClientTemplate("<a href='" + Url.Action("Retest", ControllerName) + "/#= Id #' class='k-button k-button-icontext' target='_blank'>" + Resource.Retest + "</a>").Title(Resource.Retest);
+    columns.Bound(model => model.Id);
```
Comment (rozay) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml]: Please, fix formatting. (`tab`)

File: Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml [hunk 4]
```diff
@@ -83,4 +85,14 @@
             return confirm('Сигурни ли сте, че искате да изтриете избраните ' + submissionsCount + ' решения? Моля, уверете се, че сте филтрирали коректно преди да продължите.');
         });
     });
+    function createRetestForm(id) {
+        var token = $("[name='__RequestVerificationToken']").val();
+        var formOpeningTagHtml = "<form action=\"/Administration/Submissions/Retest\" class=\"inline-form\" method=\"post\" role=\"form\">";
+        var antiforgeryTokenHtml = "<input name=\"__RequestVerificationToken\" type=hidden value=\"" + token + "\">";
```
Comment (rozay) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml]: `hidden` could be wrapped in quotes.

File: Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml [hunk 5]
```diff
@@ -83,4 +85,14 @@
...
+        var inputHtml = "<input data-val=\"true\" data-val-number=\"The field Id must be a number.\" data-val-required=\"The Id field is required.\" id=\"Id\" name=\"Id\" type=\"hidden\" value=\"" + id + "\" />";
+        var submitHtml = "<input type=\"submit\" class=\"k-button k-button-icontext\" value=Retest />";
```
Comment (rozay) [inline Open Judge System/Web/OJS.Web/Areas/Administration/Views/Submissions/_SubmissionsGrid.cshtml]: `Retest` could be wrapped in quotes.

**Pull request merged**